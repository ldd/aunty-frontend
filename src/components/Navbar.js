import React from "react";

const Navbar = () => (
  <div className="hero-head">
    <header className="navbar">
      <div className="container">
        <div className="navbar-brand">
          {/* <a className="navbar-item" href="#">
            <img
              src="https://bulma.io/images/bulma-type-white.png"
              alt="Logo"
            />
          </a> */}
          <span className="navbar-burger burger" data-target="navbarMenuHeroC">
            {/* needed to simulate a burger menu */}
            <span />
            <span />
            <span />
          </span>
        </div>
        <div id="navbarMenuHeroC" className="navbar-menu">
          <div className="navbar-end">
            <a href="/" className="navbar-item is-active">
              Home
            </a>
            <a href="/pricing" className="navbar-item">
              Pricing
            </a>
            <a href="/integrations" className="navbar-item">
              Integrations
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
);

export default Navbar;
