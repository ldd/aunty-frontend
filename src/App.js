import React, { useState } from "react";
import "./App.css";
import Navbar from "./components/Navbar";
// import Footer from "./components/Footer";
// import Home from "./components/Home"
import Pricing from "./components/Pricing";
import Home from "./components/Home";

const Router = ({ route }) => {
  if (route === "Pricing") return <Pricing />;
  // else if(route === "integrations") return <Integrations/>;
  return <Home />;
};

const App = () => {
  const [route, changeRoute] = useState("home");
  return (
    <section className="hero is-success is-fullheight">
      <Navbar route={route} changeRoute={changeRoute} />
      <Router route={route} />
      {/* <Footer /> */}
    </section>
  );
};

export default App;
