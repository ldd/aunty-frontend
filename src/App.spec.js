import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import App, { pages, Router } from "./App";
import Navbar, { NavBarLink } from "./components/Navbar";

describe("App", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders ", () => {
    const wrapper = shallow(<App />);

    const getNavbarRoute = () => wrapper.find(Navbar).prop("route");
    const getRouterRoute = () => wrapper.find(Router).prop("route");
    expect(getNavbarRoute()).toBe("");
    expect(getRouterRoute()).toBe("");

    pages.forEach((page, pageIndex) => {
      // click on a route
      wrapper
        .find(Navbar)
        .dive()
        .find(NavBarLink)
        .at(pageIndex)
        .dive()
        .simulate("click");
      expect(getNavbarRoute()).toBe(page);

      // expect such route to be rendered
      const route = wrapper.find(Router);
      expect(route.dive().name()).toBe(page);
      expect(getRouterRoute()).toBe(page);
    });
  });
});
