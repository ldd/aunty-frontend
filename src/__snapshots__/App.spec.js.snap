// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<section
  className="hero is-success is-fullheight is-bold"
>
  <div
    className="hero-head"
  >
    <header
      className="navbar"
    >
      <div
        className="container"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
        >
          <span
            className="navbar-burger burger "
            data-target="navbarMenuHeroC"
          >
            <span />
            <span />
            <span />
          </span>
        </div>
        <div
          className="navbar-menu "
          id="navbarMenuHeroC"
        >
          <div
            className="navbar-end"
          >
            <a
              className="navbar-item "
              href="#Home"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              className="navbar-item "
              href="#Pricing"
              onClick={[Function]}
            >
              Pricing
            </a>
            <a
              className="navbar-item "
              href="#Integrations"
              onClick={[Function]}
            >
              Integrations
            </a>
            <a
              className="navbar-item "
              href="#Tasks"
              onClick={[Function]}
            >
              Tasks
            </a>
            <a
              className="navbar-item "
              href="#Login"
              onClick={[Function]}
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <h1
        className="title"
      >
        Track recurrent tasks
      </h1>
      <h2
        className="subtitle"
      >
        Change 
        <strike>
          the world
        </strike>
         yourself
      </h2>
      <div
        className="container"
        style={
          Object {
            "maxWidth": 350,
          }
        }
      >
        <img
          alt="plans"
          src="/images/undraw_travel_plans_li01.svg"
        />
      </div>
    </div>
  </div>
</section>
`;
